### ISSUE TITLE
tokio: replace custom blocking tokio calls, with native tokio::runtime::Handle::block_on

### ISSUE THREAD

**Issue by michaelgrigoryan25** [Qovery/Replibyte#161]
Hi, hope you are doing well. I noticed that replibyte uses a custom function and a runtime for getting some values that need to be awaited. We can completely remove the code in [runtime.rs](https://github.com/Qovery/Replibyte/blob/main/replibyte/src/runtime.rs) since tokio gives us the ability to do the same thing natively via [tokio::runtime::Handle::block_on](https://docs.rs/tokio/latest/tokio/runtime/struct.Runtime.html#method.block_on). There are only a few things that will need to be changed in the code.

Firstly, we will need to add the `#[tokio::main]` macro and make the main function `async`.

```rs
#[tokio::main]
async fn main() {...}
```

Then, in [`s3.rs`](https://github.com/Qovery/Replibyte/blob/main/replibyte/src/datastore/s3.rs), we would change the following implementation with a custom runtime caller:

```rs
if let Some(profile) = profile {
    config_loader = config_loader
        .region(
            ProfileFileRegionProvider::builder()
                .profile_name(profile.as_ref())
                .build(),
        )
        .credentials_provider(
            ProfileFileCredentialsProvider::builder()
                .profile_name(profile.as_ref())
                .build(),
        )
        .retry_config(
            block_on(
                ProfileFileRetryConfigProvider::builder()
                    .profile_name(profile.as_ref())
                    .build()
                    .retry_config_builder(),
            )?
            .build(),
        )
}
```

with the native tokio version:

```diff
-                    block_on(
-                        ProfileFileRetryConfigProvider::builder()
-                            .profile_name(profile.as_ref())
-                            .build()
-                            .retry_config_builder(),
-                    )?
-                    .build(),
+                    tokio::runtime::Handle::current()
+                        .block_on(async move {
+                            ProfileFileRetryConfigProvider::builder()
+                                .profile_name(profile.as_ref())
+                                .build()
+                                .retry_config_builder()
+                                .await
+                        })?
+                        .build(),
                 )
```

This way, we can also get rid of additional crate dependencies, which in this case are `lazy_static` and `ctrlc` (and possibly more!). The usages of these packages are only limited to two files, so it's safe to say that replacing these should not be an issue. I consider replacing `ctrlc` with tokio's native [tokio::signal::ctrl_c](https://docs.rs/tokio/1.19.2/tokio/) function, which can be awaited in a dedicated [tokio::task](https://docs.rs/tokio/1.19.2/tokio/task/index.html) via [tokio::spawn](https://docs.rs/tokio/1.19.2/tokio/fn.spawn.html). Having fewer dependencies is always a good practice.

Thus, the new code for capturing `ctrl+c` would be:

```rs
fn wait_until_ctrlc(msg: &str) {
    let (tx, mut rx) = tokio::sync::mpsc::channel::<()>(1);
    tokio::spawn(async move {
        match tokio::signal::ctrl_c().await {
            Ok(_) => tx.send(()).await.expect("cannot capture ctrl+c"),
            Err(error) => eprintln!("{}", error),
        }
    });

    println!("{}", msg);
    tokio::runtime::Handle::current().block_on(async move { rx.recv().await });
}
```

Also notice that the channel is bounded (`channel::<()>(1)`), and can only receive one input from the sender, after which, the sending channel would become closed.

For your information, the current changeset can be found on the following pages:

- #160 (all branches linked below, depend on this PR's branch)
- https://github.com/michaelgrigoryan25/replibyte/tree/michaelgrigoryan25/tokio
- https://github.com/michaelgrigoryan25/replibyte/tree/michaelgrigoryan25/tokio-ctrlc

Let me know what you think about this proposal so that I can open a PR accordingly.

**Comment by evoxmusic** [comment]
Hi @michaelgrigoryan25 , thank you for your proposal and the great work you made here. My only concern is all the complexity this change would add for a small benefits. Can you summarize the pros and the cons of this change?

**Comment by michaelgrigoryan25** [comment]
There are no cons to this PR. Not that I can think of :slightly_smiling_face:

1. The code actually becomes less complex, since we do not need another file where we would initialize tokio's runtime. All initialization will be done by the `#[tokio::main]` macro, which will expand at compile time and provide async capabilities right out of the box.
2. Code becomes easier to read. In an `async fn` we would be able to `.await` for the desired value instead of importing the custom lazily initialized static `TOKIO_RUNTIME` variable runtime every time we need it, we call tokio's native code.
3. We can get rid of 2 dependencies (`lazy_static`, and `ctrlc`), which in all cases is a good practice. If all the functionality that we need is provided by tokio, there is no reason we would need any more dependencies. As we go down this path, and maybe remove some more crates which could be replaced in the future, we may (or may not) have a smaller binary size.
